### PR TITLE
Check all files for undefined ddoc

### PR DESCRIPTION
**Pull request by wilzbach**
Extends the undefined ddoc check for all files and fixes the violations.

### CODE REVIEW

File: dlang.org.ddoc [hunk 1]
```diff
@@ -155,6 +155,7 @@ DBLOG_LATEST_DATE2=
 DBLOG_LATEST_LINK2=https://dlang.org/blog
 DBLOG_LATEST_TITLE2=
 DMDSRC=$(HTTPS github.com/dlang/dmd/blob/master/src/dmd/$0, $0)
+DOLLAR=&dollar;
```
Comment (andralex) [inline dlang.org.ddoc]: This should go in html.ddoc

File: dlang.org.ddoc [hunk 2]
```diff
@@ -155,6 +155,7 @@ DBLOG_LATEST_DATE2=
 DBLOG_LATEST_LINK2=https://dlang.org/blog
```
Comment (andralex) [inline dlang.org.ddoc]: Pedantically this should be `$(HTTPS dlang.org/blog)`. That way if the macro ever makes it in non-html documentation, there's no problem with it.

File: faq.dd [hunk 3]
```diff
@@ -393,15 +393,15 @@ $(ITEM printf, How do I get printf() to work with strings?)
 $(CCODE
 char s[8];
 strcpy(s, "foo");
-printf("string = '$(B %s)'\n", s);
+printf("string = '%s'\n", s);
```
Comment (andralex) [inline faq.dd]: Custom escapes inside code are mightily difficult in latex, too. I'm very glad we're steering clear off them.

File: dlang.org.ddoc (hunk 2 above)
Comment (wilzbach) [reply dlang.org.ddoc]: Not related to this PR and it's only used for the front page. The reason it's defined here, is that DAutoTest doesn't error with undefined macros as these are dynamically injected.

File: dlang.org.ddoc (hunk 1 above)
Comment (wilzbach) [reply dlang.org.ddoc]: (done)

**Pull request merged**